Agent.kt

```diff
@@ -51,7 +51,7 @@ abstract class Agent<PARTITION: Partition, SESSION, CONTEXT: Context<*, *>> prot
 
     fun run(session: SESSION, partition: PARTITION, random: RandomSource): List<Report> {
         return actionHandlers[action]?.let {
-            (0 until runsPerIteration).flatMap { it(session, partition, random) }
+            (0 until runsPerIteration).flatMap { it(session, partition, random.nextSource()) }
```

Comment:
Ah. I assume this change is because, without this change, each run performs the same "random" operations?

If so, then I note that we already call `nextSource` in `Simulation.iterate`, which looks awfully redundant with this change.

Reply:
Done.